libs/components/modals/src/lib/modules/modal/modal-host.component.ts

```diff
@@ -139,11 +155,38 @@ export class SkyModalHostComponent implements OnDestroy {
       injector
     );
 
+    // modal element that was just opened
+    const modalElement = modalComponentRef.location.nativeElement;
+
     modalInstance.componentInstance = modalComponentRef.instance;
 
     this.#registerModalInstance(modalInstance);
+    // hidding all elements at the modal-host level from screenreaders
+    const hostElement = this.#elRef.nativeElement;
+    let allModals = hostElement.children;
+
+    if (SkyModalHostService.openModalCount == 1) {
+      this.hideModalHostSiblings();
+      ariaPreviousValueSiblings = this.#ariaPreviousValueSiblings;
+    }
+    if (
+      SkyModalHostService.openModalCount > 1 &&
+      allModals[allModals.length - 1] == modalElement
+    ) {
+      allModals[allModals.length - 2].setAttribute('aria-hidden', true);
+    }
 
     function closeModal() {
+      // unhide siblings if last modal is closing
+      if (SkyModalHostService.openModalCount == 1) {
+        adapter.removeAriaHidden(ariaPreviousValueSiblings);
+        ariaPreviousValueSiblings.clear();
```

Comment:
This feels like a wholistic action, so I would put this in the host's `ngOnDestroy()` method.

Reply:
that would only be called when `sky-modal-host` is destroyed right? `sky-modal-host` isn't removed from the dom even when all the modals all closed so wouldn't it make more sense for this to be called when the last modal is closed and not when host is destroyed